src/components/marketplace/order_details.tsx

```diff
@@ -70,18 +90,120 @@ class OrderDetails extends React.Component<Props, State> {
     // TODO: Refactor with hooks (needs react version update)
 
     public state = {
-        orderDetailType: OrderDetailsType.Eth,
-        zeroXPriceInUSD: new BigNumber(0),
+        limitOrder: {
+            orderDetailType: OrderDetailsType.Eth,
+            zeroXFeeInUSD: new BigNumber(0),
+            zeroXFeeInWeth: new BigNumber(0),
+            totalCostInWeth: new BigNumber(0),
+            totalCostInUSD: new BigNumber(0),
+        },
+        marketOrder: {
+            orderDetailType: OrderDetailsType.Eth,
+            zeroXFeeInUSD: new BigNumber(0),
+            zeroXFeeInWeth: new BigNumber(0),
+            totalCostInWeth: new BigNumber(0),
+            totalCostInUSD: new BigNumber(0),
+        },
+    };
+
+    public updateLimitOrderState = async () => {
+        const { tokenAmount, tokenPrice, selectedToken } = this.props;
+        if (selectedToken) {
+            /* Reduces decimals of the token amount */
+            const tokenAmountConverted = tokenAmountInUnitsToBigNumber(tokenAmount, selectedToken.decimals);
+            /* This could be refactored with promise all  */
+            const promisesArray = [getZeroXPriceInWeth(), getZeroXPriceInUSD(), getEthereumPriceInUSD()];
+            try {
+                const results = await Promise.all(promisesArray);
+                const [zeroXPriceInWeth, zeroXPriceInUSD, ethInUSD] = results;
+                /* Calculates total cost in wETH */
+                const zeroXFeeInWeth = zeroXPriceInWeth.mul(MAKER_FEE);
+                const totalPriceWithoutFeeInWeth = tokenAmountConverted.mul(tokenPrice);
+                const totalCostInWeth = totalPriceWithoutFeeInWeth.add(zeroXFeeInWeth);
+
+                /* Calculates total cost in USD */
+                const zeroXFeeInUSD = zeroXPriceInUSD.mul(MAKER_FEE);
+                const totalPriceWithoutFeeInUSD = totalPriceWithoutFeeInWeth.mul(ethInUSD);
+                const totalCostInUSD = totalPriceWithoutFeeInUSD.add(zeroXFeeInUSD);
+
+                this.setState({
+                    limitOrder: {
+                        ...this.state.limitOrder,
+                        zeroXFeeInWeth,
+                        zeroXFeeInUSD,
+                        totalCostInWeth,
+                        totalCostInUSD,
+                    },
+                });
+            } catch (error) {
+                throw error;
```

Comment:
Req: Catching and rethrowing an error is the same as not including the try/catch, as far as I know.

Reply:
I'd removed it.